core/src/main/scala/scalaz/Kleisli.scala

```diff
@@ -244,6 +244,22 @@ abstract class KleisliInstances extends KleisliInstances0 {
     new KleisliCatchable[F, A] {
       implicit def F = F0
     }
+
+  implicit def kleisliNondeterminism[F[_]: Nondeterminism, E] = new Nondeterminism[Kleisli[F, E, ?]] {
```

Comment:
- please specify explicit return type
- use [trait KleisliMonad](https://github.com/scalaz/scalaz/blob/c05cf1c4c5cc17463702b74a2de837d01d98dcc0/core/src/main/scala/scalaz/Kleisli.scala#L323) for share implementations of `bind` and `point`